src/main/java/org/broadinstitute/hellbender/engine/ProgressMeter.java

```diff
@@ -0,0 +1,200 @@
+     * clients should call one of the other constructors.
+     *
+     * @param recordsPerUpdate number of records to process before outputting a line to the logger
+     * @param timeFunction function that returns the current time in milliseconds.
+     */
+    @VisibleForTesting
+    ProgressMeter( final long recordsPerUpdate, final LongSupplier timeFunction ) {
+        Utils.nonNull(timeFunction);
+        if ( recordsPerUpdate <= 0l ) {
+            throw new IllegalArgumentException("recordsPerUpdate must be > 0");
+        }
+
+        this.recordsPerUpdate = recordsPerUpdate;
+        this.timeFunction = timeFunction;
+    }
+
+    /**
+     * Start the progress meter and produce preliminary output such as column headings.
+     */
+    public void start() {
+        logger.info("Starting traversal");
+        printHeader();
+
+        startTime = timeFunction.getAsLong();
+        currentTime = startTime;
+        numRecordsProcessed = 0l;
+        numLoggerUpdates = 0l;
+        currentLocus = null;
+    }
+
+    /**
+     * Signal to the progress meter that an additional record has been processed. Will output
+     * statistics to the logger every {@link #recordsPerUpdate} records.
+     *
+     * @param currentLocus the genomic location of the record just processed
+     */
+    public void update( final SimpleInterval currentLocus ) {
+        ++numRecordsProcessed;
+        if ( numRecordsProcessed % recordsPerUpdate == 0 ) {
+            currentTime = timeFunction.getAsLong();
+            this.currentLocus = currentLocus;
+            printProgress();
+        }
+    }
+
+    /**
+     * Stop the progress meter and output summary statistics to the logger
+     */
+    public void stop() {
+        currentTime = timeFunction.getAsLong();
+        logger.info(String.format("Traversal complete. Processed %d total records in %.1f minutes.", numRecordsProcessed, elapsedTimeInMinutes()));
+    }
+
+    /**
+     * Print column headings labelling the output from {@link #printProgress}
+     */
+    private void printHeader() {
+        logger.info(String.format("%20s  %15s  %20s  %15s",
+                                  "Current Locus", "Elapsed Minutes", "Records Processed", "Records/Minute"));
+    }
+
+    /**
+     * Output traversal statistics to the logger.
+     */
+    private void printProgress() {
+        ++numLoggerUpdates;
+        logger.info(String.format("%20s  %15.1f  %20d  %15.1f",
+                                  currentLocusString(), elapsedTimeInMinutes(), numRecordsProcessed, processingRate()));
+    }
+
+    /**
+     * @return the total minutes elapsed since we called {@link #start}
+     *
+     * This is only updated when we actually output progress and
+     * should not be called directly except in tests.
+     */
+    @VisibleForTesting
+    double elapsedTimeInMinutes() {
+        return (currentTime - startTime) / 1000.0 / 60.0;
```

Comment:
this calculation shows that the time is measures in milliseconds. Please a comment and/or change names of `currentTime` and `startTime` so say it is so. Also, I'd just extract 1000.0 * 60.0 as MILLISECONDS_PER_MINUTE to clarify